e2e-test/src/test/java/gov/cms/ab2d/e2etest/TestRunner.java

```diff
@@ -0,0 +1,571 @@
+                jobData.put("contract_id", resultSet.getInt("contract_id"));
+            }
+        }
+
+        return jobData;
+    }
+
+    private void createContract(String contractNumber) throws SQLException {
+        if(sponsorId == null) {
+            try (Connection connection = dataSource.getConnection(); Statement statement = connection.createStatement()) {
+                ResultSet resultSet = statement.executeQuery("SELECT id FROM sponsor WHERE hpms_id = 999");
+                resultSet.next();
+                sponsorId = resultSet.getInt("id");
+            }
+        }
+
+        OffsetDateTime attestationDateTime = OffsetDateTime.now();
+        try (Connection connection = dataSource.getConnection(); Statement statement = connection.createStatement()) {
+            statement.execute("INSERT INTO contract(contract_number, contract_name, " +
+                    "sponsor_id, attested_on) VALUES('" + contractNumber + "', '" + contractNumber + "', " + sponsorId + ", '" +
+                    attestationDateTime + "')");
+        }
+    }
+
+    @Test
+    public void runSystemWideExport() throws IOException, InterruptedException, JSONException, SQLException {
+        HttpResponse<String> exportResponse = exportRequest();
+
+        Assert.assertEquals(202, exportResponse.statusCode());
+        List<String> contentLocationList = exportResponse.headers().map().get("content-location");
+
+        HttpResponse<String> secondExportResponse = exportRequest();
+        Assert.assertEquals(429, secondExportResponse.statusCode());
+
+        HttpResponse<String> statusResponse = statusRequest(contentLocationList.iterator().next());
+
+        Assert.assertEquals(202, statusResponse.statusCode());
+        List<String> retryAfterList = statusResponse.headers().map().get("retry-after");
+        Assert.assertEquals(retryAfterList.iterator().next(), "5");
+        List<String> xProgressList = statusResponse.headers().map().get("x-progress");
+        Assert.assertEquals(xProgressList.iterator().next(), "0% complete");
+
+        HttpResponse<String> retryStatusResponse = statusRequest(contentLocationList.iterator().next());
+
+        Assert.assertEquals(429, retryStatusResponse.statusCode());
+        List<String> retryAfterListRepeat = retryStatusResponse.headers().map().get("retry-after");
+        Assert.assertEquals(retryAfterListRepeat.iterator().next(), "5");
+
+        String jobUUid = getJobUuid();
+
+        HttpResponse<String> deleteResponse = cancelJobRequest(jobUUid);
+        Assert.assertEquals(202, deleteResponse.statusCode());
+
+        HttpResponse<String> exportResponseSecondTry = exportRequest();
+
+        Assert.assertEquals(202, exportResponseSecondTry.statusCode());
+        List<String> contentLocationListSecondTry = exportResponseSecondTry.headers().map().get("content-location");
+
+        Thread.sleep(THREAD_DELAY);
```

Comment:
We cannot sleep for a fixed number of seconds and then expect the job to be completed since we don't know for sure for how long the job will actually run, especially when testing the external environments.

Instead, we need a _loop_ that will be checking the status periodically sleeping for the `retry-after` number of seconds in between (just to avoid 429s). The loop should exit once the status turns to done (200). The loop should fail the test if the job has not completed within a given timeframe (i.e. a timeout). For now, say, 5 minutes.